src/main/java/uk/gov/justice/probation/courtcasematcher/restclient/CourtCaseRestClient.java

```diff
@@ -31,7 +30,7 @@
 @AllArgsConstructor
 @NoArgsConstructor
 @Slf4j
-public class CourtCaseRestClient implements CourtCaseRepository {
+public class CourtCaseRestClient {
```

Comment:
Could this be 'CourtCaseServiceClient'? The class is already under the rest client package, so there is no need for 'Rest' in the name.